### ISSUE TITLE
Feature: Measurement Selected Event

### ISSUE THREAD

**Issue by mikehazell** [cornerstonejs/cornerstoneTools#1144]
I've noticed this PR which hasn't had any love from the core team yet.
https://github.com/cornerstonejs/cornerstoneTools/pull/1095

I would find this feature really useful right now. We have a UI which displays data related to an annotation in a side panel and groups annotations for a single nodule across image slices.

In some cases this UI is used in a QA process. One of our Rads has spoken about his QA workflow being to go through the DICOM slice by slice to make sure nothing was missed. While doing so he would like to be able to click on an annotation to highlight the relevant detail UI in the side panel.

I noticed in [an old issue](https://github.com/cornerstonejs/cornerstoneTools/issues/213#issuecomment-441253113) there was a hint of Annotations getting support for `isSelected` and `isHovered`. Did this ever get anywhere?

Or is there an existing way to get the functionality I'm looking for here?

**Comment by dannyrb** [comment]
Would #1095 satisfy all of your requirements? It seems like a step in the right direction.

I think our hesitancy here has been that this is a potentially hairy problem as we dig into the edge cases, and very few people are asking for a solution. 

If we reduce our scope to:

- Give annotations IDs
- Only allow one selected tool
- Make hovering smarter
  - Balance of "most recently moved" and distance
- Deprecate toolstate's `active`

We could probably do this in a generic/clean way for annotation tools.

We should be okay to merge #1095 and look at the problem in more detail if this issue has a champion.

**Comment by mikehazell** [comment]
Yes #1095 would satisfy my current requirements but I like the sound of all the things you're saying.

Because timezones, I have just finished a hacky work around so if you want to take  your time on this that's ok with me.

**Comment by jwitos** [comment]
Hi @dannyrb @mikehazell, very curious if this is still on a roadmap? Alternatively do you know a good solution to achieve this (highlighting specific annotation, connected to the information details in UI)?

**Comment by dannyrb** [comment]
@jwitos I know that annotation IDs are near future. The other enhancements are items I would like to see, but don't have any grant/client work that make them a priority. I am more than happy to volunteer time and help coordinate with a community member/partner to help scope and discuss an implementation if they would like to champion the enhancement.

**Comment by mix3d** [comment]
Whenever this effort is prioritized, I would be happy to help also add a measurement_done_modifying event as well. The current `MEASUREMENT_MODIFIED` fires too frequently while a measurement is being updated / moved, and even after throttling isn't very helpful, when all you need is the one final event.
